kubeflow/trainer/rhai/transformers.py

```
@@ -391,16 +391,125 @@ def __init__(self, cloud_remote_storage_uri: Optional[str] = None) -> None:
+                    dist.barrier()
+            except Exception as e:
+                raise RuntimeError(
+                    f"[Kubeflow] Barrier synchronization failed during checkpoint "
+                    f"{operation}: {e}. "
+                    "This typically indicates one or more training processes crashed "
+                    "or exited early. "
+                    "Check your training logs to identify which rank failed, "
+                    "verify all pods are healthy, "
+                    "and ensure distributed training is properly configured. "
+                    "Retrying the training job often resolves transient issues."
+                ) from e
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)